jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/AwsToolkitExplorerToolWindow.kt

```diff
@@ -68,52 +65,27 @@ class AwsToolkitExplorerToolWindow(private val project: Project) : SimpleToolWin
             val onTabChange = {
                 toolkitToolWindowListener.tabChanged(tabPane.getTitleAt(tabPane.selectedIndex))
             }
-            val codeWhispererTooltip = GotItTooltip("aws.toolkit.devtool.tab.whatsnew", message("codewhisperer.explorer.tooltip.comment"), project)
-                .withHeader(message("codewhisperer.explorer.tooltip.title"))
-                .withPosition(Balloon.Position.above)
-            getTabLabelComponent(DEVTOOLS_TAB_ID)?.let {
-                codeWhispererTooltip.show(it as JComponent, GotItTooltip.TOP_MIDDLE)
-            }
             tabPane.model.addChangeListener {
                 onTabChange()
             }
             onTabChange()
         }
     }
 
-    fun setDevToolsTabVisible(visible: Boolean) {
-        val index = tabPane.indexOfTab(DEVTOOLS_TAB_ID)
+    fun selectTab(tabName: String): Component? {
+        val index = tabPane.indexOfTab(tabName)
         if (index == -1) {
-            if (!visible) {
-                // if we don't need to be visible we're done
-                return
-            }
-
-            // else we need content
-            tabComponents.onEachIndexed { i, (name, contentProvider) ->
-                if (name == DEVTOOLS_TAB_ID) {
-                    tabPane.insertTab(name, null, contentProvider(), null, i)
-                    tabPane.selectedIndex = i
-                    return
-                }
-            }
+            return null
         }
 
-        // we have content
-        val content = tabPane.getComponentAt(index)
-        if (!visible) {
-            // remove if we want it gone
-            content?.let {
-                // only hide the compoenet instead of disposing all of them
-                tabPane.removeTabAt(index)
-            }
-            return
-        }
+        val component = tabPane.getComponentAt(index)
+        if (component != null) {
+            tabPane.selectedComponent = tabPane.getComponentAt(index)
 
-        // otherwise select the tab
-        content?.let {
-            tabPane.selectedIndex = index
+            return component
         }
+
+        return null
```

Comment:
We won't get this null case but are we handling it anywhere on the off chance that we do?

Reply:
no, let's handle when a consumer actually cares about the result